tests/components/hue/test_config_flow.py

```diff
@@ -76,21 +83,104 @@ async def test_flow_works(hass):
     assert result["data"] == {
         "host": "1.2.3.4",
         "username": "home-assistant#test-home",
-        "allow_hue_groups": False,
     }
 
     assert len(mock_bridge.initialize.mock_calls) == 1
 
 
-async def test_flow_no_discovered_bridges(hass, aioclient_mock):
+async def test_manual_flow_works(hass, aioclient_mock):
+    """Test config flow discovers only already configured bridges."""
+    mock_bridge = get_mock_bridge()
+
+    with patch(
+        "homeassistant.components.hue.config_flow.discover_nupnp",
+        return_value=[mock_bridge],
+    ):
+        result = await hass.config_entries.flow.async_init(
+            const.DOMAIN, context={"source": "user"}
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "init"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"id": "manual"}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual"
+
+    bridge = get_mock_bridge(
+        bridge_id="id-1234", host="2.2.2.2", username="username-abc"
+    )
+
+    with patch(
+        "aiohue.Bridge", return_value=bridge,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "2.2.2.2"}
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "link"
+
+    with patch("homeassistant.components.hue.config_flow.authenticate_bridge"), patch(
+        "homeassistant.components.hue.async_unload_entry", return_value=True
+    ):
+        result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == "Mock Bridge"
+    assert result["data"] == {
+        "host": "2.2.2.2",
+        "username": "username-abc",
+    }
+    entries = hass.config_entries.async_entries("hue")
+    assert len(entries) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5795327)